subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -535,18 +528,24 @@ void nrf_cloud_run(void)
 	atomic_set(&transport_disconnected, 0);
 
 	while (true) {
-		ret = poll(fds, ARRAY_SIZE(fds), POLL_TIMEOUT_MS);
+		ret = poll(fds, ARRAY_SIZE(fds),
+			   api_keepalive_time_left(nrf_cloud_backend));
 
+		/* If poll returns 0 the timeout has expired. */
 		if (ret == 0) {
-			if (cloud_keepalive_time_left(nrf_cloud_backend) <
-			    POLL_TIMEOUT_MS) {
-				api_ping(nrf_cloud_backend);
-			}
+			api_ping(nrf_cloud_backend);
 			continue;
 		}
 
 		if ((fds[0].revents & POLLIN) == POLLIN) {
 			api_input(nrf_cloud_backend);
+
+			if (atomic_get(&transport_disconnected) == 1) {
+				LOG_DBG("The cloud socket is already closed.");
+				cloud_evt.data.err = CLOUD_DISCONNECT_MISC;
```

Comment:
this line has no effect and can be removed.

Reply:
ah, I see. Removed.